src/main/java/sirius/biz/storage/layer2/L3Uplink.java

```diff
@@ -130,6 +134,8 @@ protected VirtualFile createPlaceholder(@Nullable Directory parentDirectory, Vir
 
             MutableVirtualFile file = MutableVirtualFile.checkedCreate(parent, name);
             file.attach(new Placeholder(parent, name));
+            parent.tryAs(BlobStorageSpace.class)
+                  .ifPresent(blobStorageSpace -> file.attach(BlobStorageSpace.class, blobStorageSpace));
```

Comment:
```suggestion
            file.attach(BlobStorageSpace.class, parentDirectory.getStorageSpace());
```

Reply:
Won't work because in this path: `/work/existing-folder/foo/bar`
when we `bar`, parentDirectory will be null since its parent is a PlaceHolder. This is why we always inherit from the parent. The first placeholder gets it from a real Directory, all others down the chain shall get it.